plugins/floodProtectionV2.cjs

```diff
@@ -0,0 +1,58 @@
+/*
+    Simple flood protection plugin.
+    Note: Received Telegram message time accuracy is one second.
+*/
+
+const userList = {};
+
+const isFlooding = (fromId, now, interval) => {
+  const user = userList[fromId];
+  const diff = now - user.lastTime;
+  user.lastTime = now;
+
+  if (diff <= interval) {
+    if (!user.warned) {
+      user.warned = true;
+    }
+    return true;
+  }
+  user.warned = false;
+  return false;
+};
+
+// Export bot module
+module.exports = {
+  id: "floodProtectionV2",
+  defaultConfig: {
+    interval: 5, // segundos
+    message: "Too many messages, relax!",
+  },
+
+  plugin(bot, pluginConfig) {
+    const interval = Number(pluginConfig.interval) || 1;
+    const text = pluginConfig.message;
+
+    bot.mod("message", (data) => {
+      const { message: msg } = data;
+      const fromId = msg.from.id;
+      const now = new Date(msg.date);
+      const user = userList[fromId];
+      const { media_group_id: mediaGroupId } = msg;
+
+      const isAlbum = !!mediaGroupId;

```

Comment:
```suggestion
      const isAlbum = Boolean(mediaGroupId);
```

O not not as vezes pode dar valores inesperados, o Boolean resolve a maioria dos casos. O pessoal nessa thread discute bastante sobre isso: https://stackoverflow.com/questions/784929/what-is-the-not-not-operator-in-javascript